zcash_client_backend/src/data_api.rs

```diff
@@ -1229,14 +1225,16 @@ pub trait WalletRead {
         ufvk: &UnifiedFullViewingKey,
     ) -> Result<Option<Self::Account>, Self::Error>;
 
-    /// Returns the most recently generated unified address for the specified account, if the
-    /// account identifier specified refers to a valid account for this wallet.
+    /// Returns the most recently generated unified address for the specified account that conforms
+    /// to the specified address filter, if the account identifier specified refers to a valid
+    /// account for this wallet.
     ///
-    /// This will return `Ok(None)` if the account identifier does not correspond to a known
-    /// account.
-    fn get_current_address(
+    /// This will return `Ok(None)` if no previously generated address conforms to the specified
+    /// request.
+    fn get_last_generated_address(
```

Comment:
```suggestion
    fn get_last_generated_address_matching(
```
I agree with this suggestion of @str4d's, independent of whether we also add `get_last_sequential_diversifier_index` and/or `get_last_time_based_diversifier_index`.